### PR TITLE
Washing machine cleanup & removes crayon/stamp painting

### PR DESCRIPTION
**Pull request by tkdrg**
- **Removes** crayon/stamp painting.

I'll talk a little more about this one because I know it will be controversial. Yes, it's a feature. Yes, I want to remove it. The code is terrible, but I could have built a new system for allowing this. My reasoning for not doing it was that it isn't worth it to spend my time building a system for such a rarely used feature.
- Rewrites washing machines to be clearer and simpler.
- Washing machines no longer gib corgis (this is redundant now that we can build meat spikes).
- Removes item_color\* and suit_color vars (redundant).
- Renamed some icon_states to allow for that.

*item_color was kept as a higher level var in some situations where it isn't actually redundant (namely: eswords, hardsuits, softcaps, paint, implants)

:cl:
rscdel: You can no longer paint items by using crayons or stamps in a washing machine.
rscdel: Washing machines no longer gib corgis. You can build meat spikes instead.
/:cl:

### CODE REVIEW

**Comment by CosmicScientist** [comment]
:+1: :-1: I need a halfway emote.

**Comment by Anonus** [comment]
This is horrible as a change but excellent code.

File: code/game/machinery/washing_machine.dm [hunk 1]
```diff
@@ -2,307 +2,66 @@
...
-	else if(istype(W,/obj/item/stack/sheet/hairlesshide) || \
-		istype(W,/obj/item/clothing/under) || \
-		istype(W,/obj/item/clothing/mask) || \
-		istype(W,/obj/item/clothing/head) || \
-		istype(W,/obj/item/clothing/gloves) || \
-		istype(W,/obj/item/clothing/shoes) || \
-		istype(W,/obj/item/clothing/suit) || \
-		istype(W,/obj/item/weapon/bedsheet))
-
-		//YES, it's hardcoded... saves a var/can_be_washed for every single clothing item.
```
Comment (Anonus) [inline code/game/machinery/washing_machine.dm]: This comment oh god.

Comment (tkdrg) [inline code/game/machinery/washing_machine.dm]: What if erro had actually done it

**Comment by RemieRichards** [comment]
@tkdrg jumpsuit rolling uses ```item_color```, does that still work?

**Comment by tkdrg** [comment]
Yes, I offloaded that to `update_icon` instead.

**Comment by RemieRichards** [comment]
So you did, I must have missed that.

**Comment by optimumtact** [comment]
Woah Woah Woah Woah.

It wouldn't be that hard to have a datum containing a list of type colour -> new type transforms

Why the big rush to remove clothes painting? I use it all the time to hide my electrical gloves

**Comment by Supermichael777** [comment]
FUN REMOVED :<

**Comment by Incoming5643** [comment]
:-1: Literally no other way in the game to get grey gloves

**Comment by Robustin** [comment]
I'm no rocket surgeon but what loss is there regarding "bad code" for some rarely used feature?

These are great features being cut because they're powered by ugly code?

**Comment by Cheridan** [comment]
what's the point of washing machines after these changes? if you can't recolor things with them then they're strictly inferior to just walking under a shower. 

they'll have no point so the better solution instead of merging this would just be removing them from the map

**Comment by Core0verload** [comment]
:-1: Rarely used? Fuck you.

**Comment by Firecage** [comment]
...Dislike this PR, sorry :-1: 

And I have to agree with Cheridan. if you remove this feature, you can just aswell remove washing machines from the code and the map, since they will literally have been made useless.

**Comment by duncathan** [comment]
I support the code improvements but not the feature removal
You can make color changing good, I believe in you

**Comment by tkdrg** [comment]
I guess I could, but like I said, I'd rather not spend my time on it. Since it looks like this won't be merged, I'll close it and leave the branch if anybody wants to finish it.

**Pull request closed without merging**